sql/src/main/java/io/crate/execution/engine/pipeline/MapRowUsingInputs.java

```diff
@@ -22,70 +22,49 @@
 
 package io.crate.execution.engine.pipeline;
 
-import io.crate.data.BatchIterator;
-import io.crate.data.ForwardingBatchIterator;
 import io.crate.data.Input;
 import io.crate.data.Row;
 import io.crate.execution.engine.collect.CollectExpression;
 import io.crate.expression.InputRow;
 
 import java.util.List;
 import java.util.RandomAccess;
+import java.util.function.Function;
 
 /**
- * BatchIterator implementation that can transform rows using {@link Input}s and {@link CollectExpression}s.
+ * Function that transforms a Row using the supplied {@link Input}s and {@link CollectExpression}s.
  *
  * Example:
  * <pre>
- *     BatchIterator source data:
- *      [ 1, 2, 3 ]
- *
  *     Input: add(inputColumn(0), 2)
- *     expressions: InputColumnExpression
+ *     Expressions: InputColumnExpression
  *
+ *     input:
+ *      Row[1]
  *     output:
- *      [ 3, 4, 5 ]
+ *      Row(3)
```

Comment:
I'm assuming this means a Row with 3 columns?

Reply:
No it's a row with a single column with the value 3. I'll change it to square brackets to match the input example